File: 10/02/PageObjects/DynamicLoadingPage.cs
using OpenQA.Selenium;

namespace PageObjects 
{
    class DynamicLoadingPage : BasePage
    {
        By StartButton = By.CssSelector("#start > button");
        By FinishText = By.Id("finish");

        public DynamicLoadingPage(IWebDriver Driver) : base(Driver) { }

        public void LoadExample(int ExampleNumber)
        {
            Visit("http://the-internet.herokuapp.com/dynamic_loading/" + ExampleNumber);
            Click(StartButton);
        }

        public bool FinishTextPresent()
        {
            return IsDisplayed(FinishText, 10);
        }
    }
}

File: code-examples/python/10/02/pages/dynamic_loading_page.py
from selenium.webdriver.common.by import By
from . base_page import BasePage


class DynamicLoadingPage(BasePage):
    _start_button = {"by": By.CSS_SELECTOR, "value": "#start button"}
    _finish_text = {"by": By.ID, "value": "finish"}

    def ___init___(self, driver):
        self.driver = driver

    def load_example(self, example_number):
        self._visit(
            "http://the-internet.herokuapp.com/dynamic_loading/" +
            example_number)
        self._click(self._start_button)

    def finish_text_present(self):
        return self._is_displayed(self._finish_text, 10)
